airflow/configuration.py

```diff
@@ -197,15 +186,20 @@ def get(self, section, key, **kwargs):
             return option
 
         # ...then the config file
-        if self.has_option(section, key):
+        if super(AirflowConfigParser, self).has_option(section, key):
```

Comment:
This seems an odd thing to do -- can we not just use `self.has_option(section, key)` here? If not expand in the comment as to why please.

Reply:
Yeah, so as `self.has_option` uses `self.get` we would get infinite recursion here. Also self.has_option also would get defaults options already here. I've added comments to the code to address that. 

We could reimplement has_option but it would not be usable here anyway as it would look not only at the config file but env variables and the default config. It would then have to have some flag to avoid that.